src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/InterfaceDispatchCellNode.cs

```diff
@@ -13,28 +13,55 @@ namespace ILCompiler.DependencyAnalysis
     class InterfaceDispatchCellNode : ObjectNode, ISymbolNode
     {
         MethodDesc _targetMethod;
+        string _callSiteIdentifier;
 
-        public InterfaceDispatchCellNode(MethodDesc targetMethod)
+        public InterfaceDispatchCellNode(MethodDesc targetMethod, string callSiteIdentifier)
         {
             Debug.Assert(targetMethod.OwningType.IsInterface);
             Debug.Assert(!targetMethod.IsSharedByGenericInstantiations);
             _targetMethod = targetMethod;
+            _callSiteIdentifier = callSiteIdentifier;
         }
 
         public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         {
-            sb.Append("__InterfaceDispatchCell_");
-            sb.Append(NodeFactory.NameMangler.GetMangledMethodName(_targetMethod));
+            sb.Append(GetMangledName(_targetMethod, _callSiteIdentifier));
         }
         public int Offset => 0;
-        public override bool IsShareable => true;
+
+        public override bool IsShareable => false;
+
+        public static string GetMangledName(MethodDesc method, string callSiteIdentifier)
+        {
+            string name = NodeFactory.NameMangler.CompilationUnitPrefix + "__InterfaceDispatchCell_" + NodeFactory.NameMangler.GetMangledMethodName(method);
+
+            if (!string.IsNullOrEmpty(callSiteIdentifier))
+            {
+                name += "_" + callSiteIdentifier;
+            }
+
+            return name;
+        }
 
         protected override string GetName() => this.GetMangledName();
 
         public override ObjectNodeSection Section => ObjectNodeSection.DataSection;
 
         public override bool StaticDependenciesAreComputed => true;
 
+        // The second cell field uses the two lower-order bits to communicate the contents.
+        // See src\Native\Runtime\inc\rhbinder.h
+        private enum CachePointerType
+        {
+            // The low 2 bits of the m_pCache pointer are treated specially so that we can avoid the need for 
+            // extra fields on this type.
+            CachePointerIsInterfaceRelativePointer = 0x3,
+            CachePointerIsIndirectedInterfaceRelativePointer = 0x2,
+            CachePointerIsInterfacePointer = 0x1,
```

Comment:
David renamed this one recently - we should re-sync this with names in rhbinder.h.

@davidwrighton Do we have any expectations of this being used from any other managed code? Should this be defined somewhere under `Common\src\Internal\Runtime` (so that we don't end up with a dozen copies as usual)?